arch/powerpc/platforms/pseries/setup.c

```diff
@@ -455,6 +455,139 @@ static void __init find_and_init_phbs(void)
 	of_pci_check_probe_only();
 }
 
+#ifdef CONFIG_PCI_IOV
+resource_size_t pseries_get_iov_fw_values(struct pci_dev *dev, int resno,
+			  int value)
+{
+	struct vf_bar_wdw {
+		__be64  addr;
+		__be64  aperture_size;
+		__be64  wdw_size;
+	};
+
+	struct vf_bar_wdw window_avail[PCI_SRIOV_NUM_BARS];
+	const int *indexes;
+	struct device_node *dn = pci_device_to_OF_node(dev);
+	int i, r, num_res;
+	resource_size_t return_value;
+
+	indexes = of_get_property(dn, "ibm,open-sriov-vf-bar-info", NULL);
+	if (indexes == NULL)
+		return  0;
+
+	memset(window_avail,
+	       0, sizeof(struct vf_bar_wdw)*PCI_SRIOV_NUM_BARS);
+	return_value = 0;
+	/*
+	 * First element in the array is the number of Bars
+	 * returned.  Search through the list to find the matching
+	 * bar
+	 */
+	num_res = of_read_number(&indexes[0], 1);
+	for (i = 2, r = 0; r < num_res
+	     && r < PCI_SRIOV_NUM_BARS; i += 7, r++) {
+
+		window_avail[r].addr = of_read_number(&indexes[i], 2);
+		window_avail[r].aperture_size =
+			of_read_number(&indexes[i + 2], 2);
+		window_avail[r].wdw_size = of_read_number(&indexes[i + 4], 2);
+	}
+
+	if (value == 1)
+		return_value = window_avail[resno].addr;
+	else if (value == 2)
+		return_value = window_avail[resno].aperture_size;
+	else if (value == 3)
+		return_value = window_avail[resno].wdw_size;
+
+	return  return_value;
+}
+
+int of_pci_parse_iov_addrs(struct pci_dev *dev)
+{
+	struct device_node *dn = pci_device_to_OF_node(dev);
+	struct pci_bus_region region;
+	struct resource *res, *root, *conflict;
+	resource_size_t base, size, start, end;
+	int i, r, num_res;
+	const int *indexes;
+
+	indexes = of_get_property(dn, "ibm,open-sriov-vf-bar-info", NULL);
+	if (indexes == NULL)
+		return -ENOMEM;
+	/*
+	 * First element in the array is the number of Bars
+	 * returned.  Search through the list to find the matching
+	 * bars assign them from firmware into resources structure.
+	 */
+	num_res = of_read_number(&indexes[0], 1);
+	for (i = 2, r = 0; r < num_res
+	     && r < PCI_SRIOV_NUM_BARS; i += 7, r++) {
```

Comment:
Same here with all these values, what is 7? When you define them use them throughout your code in this function and the previous.

Reply:
will define as next_entry. thanks.